client/src/components/shell/navbar/NavbarUser.jsx

```diff
@@ -1,30 +1,51 @@
-import React from "react";
-import { Box, Image, Text } from "@mantine/core";
+import React, { useState } from "react";
+import { Anchor, Avatar, Box, Text } from "@mantine/core";
 
 import { useNavbarStyles } from "../../../hooks/styles/use-navbar-styles";
 import { useMediaQuery } from "@mantine/hooks";
+import SignUpModal from "../../homepage/SignUpModal";
 
 const NavbarUser = () => {
   const { classes } = useNavbarStyles();
+  const [opened, setOpened] = useState(false);
+  const isGuest = window.localStorage.getItem("userType") === "guest";
```

Comment:
I'd change this to `isUser` and check for "user" in localStorage. That way we will only display Avatar for logged in users and cover other cases like "guest" and `undefined` with the text placeholder.